mmdet/models/dense_heads/ascend_ssd_head.py

```diff
@@ -0,0 +1,242 @@
+                     type='SSDAnchorGenerator',
+                     scale_major=False,
+                     input_size=300,
+                     strides=[8, 16, 32, 64, 100, 300],
+                     ratios=([2], [2, 3], [2, 3], [2, 3], [2], [2]),
+                     basesize_ratio_range=(0.1, 0.9)),
+                 bbox_coder=dict(
+                     type='DeltaXYWHBBoxCoder',
+                     clip_border=True,
+                     target_means=[.0, .0, .0, .0],
+                     target_stds=[1.0, 1.0, 1.0, 1.0],
+                 ),
+                 reg_decoded_bbox=False,
+                 train_cfg=None,
+                 test_cfg=None,
+                 init_cfg=dict(
+                     type='Xavier',
+                     layer='Conv2d',
+                     distribution='uniform',
+                     bias=0)):
+        super(AscendSSDHead, self).__init__(
+            num_classes,
+            in_channels,
+            stacked_convs,
+            feat_channels,
+            use_depthwise,
+            conv_cfg,
+            norm_cfg,
+            act_cfg,
+            anchor_generator,
+            bbox_coder,
+            reg_decoded_bbox,
+            train_cfg,
+            test_cfg,
+            init_cfg)
+
+    def get_static_anchors(self, featmap_sizes, img_metas, device="cuda"):
+        if not hasattr(self, 'static_anchors') or not hasattr(self, 'static_valid_flags'):
+            static_anchors, static_valid_flags = self.get_anchors(featmap_sizes, img_metas, device)
+            self.static_anchors = static_anchors
+            self.static_valid_flags = static_valid_flags
+        return self.static_anchors, self.static_valid_flags
+
+    def get_targets(self,
+                    anchor_list,
+                    valid_flag_list,
+                    gt_bboxes_list,
+                    img_metas,
+                    gt_bboxes_ignore_list=None,
+                    gt_labels_list=None,
+                    label_channels=1,
+                    unmap_outputs=True,
+                    return_sampling_results=False,
+                    return_level=True):
+        return AscendAnchorHead.get_targets(
+            self,
+            anchor_list,
+            valid_flag_list,
+            gt_bboxes_list,
+            img_metas,
+            gt_bboxes_ignore_list,
+            gt_labels_list,
+            label_channels,
+            unmap_outputs,
+            return_sampling_results,
+            return_level,
+        )
+
+    def concat_loss(self,
+                    concat_cls_score, concat_bbox_pred,
+                    concat_anchor, concat_labels,
+                    concat_label_weights,
+                    concat_bbox_targets, concat_bbox_weights,
+                    concat_pos_mask, concat_neg_mask,
+                    num_total_samples):
+        num_images, num_anchors, _ = concat_anchor.size()
+
+        concat_loss_cls_all = F.cross_entropy(
```

Comment:
can we use self.cls_loss to calculate the loss? Otherwise, the argument/config will have no effect.

Reply:
Keep consistent with the original code. The argument/config have no effect in ssd loss function.